File: Stack.cpp
#include <cstdio>
#include <iostream>


template<typename T>
class StackNode{
public:
    T data;
    StackNode *next;
    StackNode *prev;

    StackNode(T value){
        this->data = value;
    }

};


class Stack{
private: 
    StackNode<int> *last;

public:
    Stack(){
        this->last = nullptr;
    };

    void push (int value){
        StackNode<int> *valueAdded = new StackNode<int>(value);
        if(this->last == nullptr){
            this->last = valueAdded;
        };
        this->last->next = valueAdded;
        valueAdded->prev = this->last;
        this->last = this->last->next;
    };

    void show(){
        /*return this->last->data;*/
        std::cout << this->last->data;
    };

    void pop(){
        if(this->last == nullptr){
            std::cout << "Stack vazia";
        };
        this->last = this->last->prev;
        this->last->next = nullptr;
    };


};

File: datastructuresInPython/Stack.py
class StackNode:
    next = None
    data = None
    prev = None
    def __init__(self,dataValue):
        self.data = dataValue

class Stack:
    last = None
    def __init__(self):
        ...

    def push(self,value):
        valueToBeAdded = StackNode(value)
        if(self.last is None):
            self.last = valueToBeAdded
        self.last.next = valueToBeAdded
        valueToBeAdded.prev = self.last
        self.last = self.last.next
    
    def show(self):
        if(self.last is StackNode(None)):
            print("none")
        print(self.last.data)

    def pop(self):
        if(self.last.data is None):
            return ...
        self.last = self.last.prev
        #self.last.next = None


if __name__ == "__main__":

    st = Stack()
    st.push(3)
    st.push(90)
    st.push(67)
    st.push(0)
    st.push(54)
    st.push(104)
    st.pop()
    st.pop()
    st.pop()
    st.pop()
    st.pop()
    st.pop()
    st.pop()
    st.pop()
    st.pop()
    st.show()